### PR TITLE
Feature/login backend

### LINKED ISSUE #3: Login / Register Backend
This issue contains the AccountData and AccountController class.

AccountData:
This class can fetch a password if provided a username from the database and can add a new account to the database.

AccountController:
This class can check whether or not account credentials are correct and can register a new user.

### PR DESCRIPTION
**Pull request by Martijnpold**
### Proposed changes
- Add AccountData class to fetch passwords and register new users.
- Add AccountController class for general account checks.
- Add UserTakenException for taken usernames.

Closes #3

### CODE REVIEW

**Comment by Martijnpold** [comment]
@yymar You might be interested in reviewing this, since you made the login screen.

File: src/catan/controllers/account/UserTakenException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package catan.controllers.account;
+
+/**
+ * Thrown by AccountController is a registered user is already taken.
```
Comment (yymar) [inline src/catan/controllers/account/UserTakenException.java]: Thrown by AccountController if**?

File: src/catan/model/AccountData.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class AccountData {
+    public String fetchPassword(String user) {
+        try {
+            ResultSet rs = SQL.getInstance().quickExecute(
+                    "SELECT `password` " +
+                            "FROM `accountdata` " +
+                            "WHERE `username` = '" + user + "'"
+            );
+            if (rs.next()) {
```
Comment (yymar) [inline src/catan/model/AccountData.java]: I personally don't get this, just wondering what `rs.next()` does.

**Comment by Martijnpold** [comment]
@yymar

**Pull request merged**